### PR TITLE
SONAR-10313 change route and key of issue to issues id instead of key

### PR DESCRIPTION
**Pull request by sns-seb**
… key

### CODE REVIEW

File: server/sonar-server/src/main/java/org/sonar/server/rule/index/RuleIndex.java [hunk 1]
```diff
@@ -183,8 +182,9 @@ public RuleIndex(EsClient client, System2 system2) {
     }
 
     esSearch.setQuery(boolQuery().must(qb).filter(fb));
+    System.out.println(esSearch);
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/rule/index/RuleIndex.java]: Code Smell Code Smell: Replace this use of System.out or System.err by a logger. ([squid:S106](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS106))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.sonarqube%3Asonarqube&branch=feature%2Fsl%2F7.1%2FSONAR-10313%2Fremove_key_from_rule_indexes&issues=AWFL-JqXNbtQu1yJ8qeE&open=AWFL-JqXNbtQu1yJ8qeE)

[ISSUE_KEY]: AWFL-JqXNbtQu1yJ8qeE

File: server/sonar-server/src/main/java/org/sonar/server/rule/RuleCreator.java [hunk 2]
```diff
@@ -151,11 +147,15 @@ private static void validateRuleKey(List<String> errors, String ruleKey) {
     }
   }
 
-  private Optional<RuleDefinitionDto> loadRule(RuleKey ruleKey, DbSession dbSession) {
+  private Optional<RuleDefinitionDto> loadRule(DbSession dbSession, int ruleId) {
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/rule/RuleCreator.java]: Code Smell Code Smell: Remove this unused private "loadRule" method. ([squid:UnusedPrivateMethod](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AUnusedPrivateMethod))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.sonarqube%3Asonarqube&branch=feature%2Fsl%2F7.1%2FSONAR-10313%2Fremove_key_from_rule_indexes&issues=AWFL-Jp4NbtQu1yJ8qeD&open=AWFL-Jp4NbtQu1yJ8qeD)

[ISSUE_KEY]: AWFL-Jp4NbtQu1yJ8qeD

File: server/sonar-server/src/main/java/org/sonar/server/qualityprofile/RuleActivationContext.java [hunk 3]
```diff
@@ -66,6 +66,7 @@
 
   // the rules
   private final Map<RuleKey, RuleWrapper> rulesByKey;
```
[the remark this answers is not in the document]
Comment (sns-seb) [reply server/sonar-server/src/main/java/org/sonar/server/qualityprofile/RuleActivationContext.java]: there is still some place where the key is used.
I'll try and replace those by the rule it

**Comment by sns-seb** [comment]
@ehartmann feedback applied

**Pull request merged**